src/compiler/checker.ts

```diff
@@ -17906,16 +17906,19 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             accessNode;
     }
 
-    function isPatternLiteralPlaceholderType(type: Type): boolean {
+    function isPatternLiteralPlaceholderType(type: Type, ignoreGenericIntersections = false): boolean {
         if (type.flags & TypeFlags.Intersection) {
-            return some((type as IntersectionType).types, t => !!(t.flags & (TypeFlags.Literal | TypeFlags.Null | TypeFlags.Undefined)) || isPatternLiteralPlaceholderType(t));

```

Comment:
I think this entire fix can be simplified to simply adding a check that the type isn't generic:

```ts
return !isGenericType(type) && some(...);
```

Generic types should never be classified as placeholders since upon instantiation they may become something completely different.

Reply:
Oh, that's cool! Thanks for the tip. I was worried that `addSpans` would fail if I did something like this. It seems that (somewhat confusingly) `isGenericIndexType` already returns `true` for an intersection like this so `addSpans` is covered.